src/ProductConstructionService/ProductConstructionService.ReproTool/ReproTool.cs

```diff
@@ -0,0 +1,280 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Maestro.Data;
+using Microsoft.DotNet.DarcLib;
+using Microsoft.DotNet.ProductConstructionService.Client;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Logging;
+using Octokit;
+using Build = Microsoft.DotNet.ProductConstructionService.Client.Models.Build;
+using BuildData = Microsoft.DotNet.ProductConstructionService.Client.Models.BuildData;
+using GitHubClient = Octokit.GitHubClient;
+
+namespace ProductConstructionService.ReproTool;
+
+internal class ReproTool(
+    IBarApiClient barClient,
+    ReproToolOptions options,
+    BuildAssetRegistryContext context,
+    DarcProcessManager darcProcessManager,
+    IProductConstructionServiceApi pcsApi,
+    GitHubClient ghClient,
+    ILogger<ReproTool> logger)
+{
+    private const string MaestroAuthTestOrgName = "maestro-auth-test";
+    private const string VmrForkRepoName = "dotnet";
+    private const string VmrForkUri = $"https://github.com/{MaestroAuthTestOrgName}/{VmrForkRepoName}";
+    private const string ProductRepoFormat = $"https://github.com/{MaestroAuthTestOrgName}/";
+    private const long InstallationId = 289474;
+    private const string SourceMappingsPath = "src/source-mappings.json";
+    private const string SourceManifestPath = "src/source-manifest.json";
```

Comment:
Can't we just get this from `DarcLib`?

Reply:
fixed, had to make some small tweeks in VmrInfo. I think there were duplicate static variables there with the same value, so I deleted one, please correct me if I missed something